pkg/frontend/internal_executor.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package frontend
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/config"
+	ie "github.com/matrixorigin/matrixone/pkg/util/internalExecutor"
+	"github.com/matrixorigin/matrixone/pkg/vm/mmu/guest"
+)
+
+func applyOverride(sess *Session, opts ie.SessionOverrideOptions) {
+	if opts.Database != nil {
+		sess.protocol.SetDatabaseName(*opts.Database)
+	}
+
+	if opts.Username != nil {
+		sess.protocol.SetUserName(*opts.Username)
+	}
+
+	if opts.IsInternal != nil {
+		sess.IsInternal = *opts.IsInternal
+	}
+}
+
+type internalExecutor struct {
```

Comment:
Are all internal sqls executed by one background internalExecutor?

Reply:
If sending sql by Request, it may not be convient to set sepcial field, like IsInternal, in Session. Anyway, this seems not a minor change, maybe it can be considered further when the internalExecutor needs to support select